mapknowledge/nposparql.py

```diff
@@ -244,32 +332,158 @@ def __metadata(self, neuron):
+                    return []
+                elif len(new_node[0]) == 0:
+                    new_node = [new_node[1][0], tuple(list(new_node[1])[1:])]
+                edge += [tuple(new_node)]
+            return tuple(edge)
+
+        self.__apinat_connectivities = {}
+        for partial_order in partial_order_text.split("\n\n"):
+            if "neuronPartialOrder" in partial_order:
+                neuron, nested_structure = partial_order.split(
+                    "ilxtr:neuronPartialOrder"
+                )
+                nested_structure = nested_structure.replace(".", "")
+                # replace consecutive space with a single space
+                nested_structure = re.sub(
+                    r"\s+", " ", nested_structure).strip()
+                # adding coma
+                pattern = r"\[([^]]+)\]"
+
+                def add_comma(match):
+                    elements = match.group(1).strip().split()
+                    return "[" + ", ".join(elements) + "]"
+
+                nested_structure = re.sub(pattern, add_comma, nested_structure)
+                # Quoting ILX and UBERON
+                pattern = r"(ILX:\d+|UBERON:\d+)"
+                nested_structure = re.sub(pattern, r'"\1"', nested_structure)
+                # Specifying tuple
+                nested_structure = nested_structure.replace(" )", ", )").replace(
+                    " ( ", ", ( "
+                )
+                # convert to tuple
+                conn_structure = ast.literal_eval(nested_structure)
+                # parse connectivities
+                connectivities = []
+                if conn_structure != "blank":
+                    if len(conn_structure) > 1:
+                        root = (
+                            (
+                                list(reversed(conn_structure[0]))[0],
+                                tuple(list(reversed(conn_structure[0]))[1:]),
+                            )
+                            if isinstance(conn_structure[0], list)
+                            else (conn_structure[0], ())
+                        )
+                        parse_connectivities(
+                            connectivities, conn_structure[1:], root)
+                # filter connectivities based on EXCLUDE_LAYERS
+                filtered_connectivities = []
+                for c in connectivities:
+                    edge = filter_layer(c)
+                    if len(edge) > 0:
+                        if edge[0] != edge[1]:
+                            filtered_connectivities += [edge]
+                self.__apinat_connectivities[neuron.strip()] = filtered_connectivities
+
+    def connectivity_models(self):
+        models = {}
+        for rst in self.__connectivity_models():
+            models[rst['Model_ID']] = {"label": "", "version": ""}
+        return models
+
+    def npo_released(self):
+        return self.__db_version()['NPO']
+
```

Comment:
Why `npo_released` as the method name? This is part of the `NpoSparql` class so the `npo_` prefix is irrelevant.

How about `version` as the name? (And in the `SciCrunch` class, I will rename `sckan_build` to `sckan_version` for consistency (the `sckan_` prefix is because I saw SciCrunch as being more than SCKAN, but this may now have changed -- will check with K-Core).